tagController.js

```diff
@@ -138,49 +186,64 @@ function tagRowSelectUpdate(tagRowId){
     let tagCost = document.getElementById("tagRow_" + tagRowId + "_costBox");
     let tagDesc = document.getElementById("tagRow_" + tagRowId + "_desc");
     let tagRanks = document.getElementById("tagRow_" + tagRowId + "_rank");
+    let tagId = document.getElementById("tagRow_" + tagRowId + "_tagId");
+    let tagRowCheck = document.getElementById('tagRow_' + tagRowId + "_checkBox");
 
     if(tagSelector.value > 0){
         var tagData = tagDataLocalList().tags[tagSelector.value];
+
+        if(tagData.id){
```

Comment:
I notice you're writing `if(tagData.id){` a lot. What is that actually checking? Do you ever expect it to be undefined?

Reply:
I've had bug cases where its undefined, they happen less frequently now, but yes.